src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,263 @@
+            _innerUnlink.call(_fs, pathname, function(err){
+                // only update cache once original was successfull
+                if (!err){
+                    delete _cache[pathname];
+                }
+                callback(err);
+            });
+        };
+
+        // original writeFile
+        var _innerWriteFile = _fs.writeFile;
+        // overwrite original writeFile and add bookkeeping of project state and call original writeFile.
+        _fs.writeFile = function(filename, data, options, callback){
+            if (typeof options === "function"){
+                callback = options;
+                options = {};
+            }
+
+            _innerWriteFile.call(_fs, filename, data, function(err){
+                // only update cache once original was successfull
+                if (!err){
+                    _cache[filename] = data.length;
+                }
+                callback(err);
+            });
+        };
+
+        // original rename
+        var _innerRename = _fs.rename;
+        // overwrite original rename and add bookkeeping of project state and call original rename.
+        // this is essential because we don't want to lose track of file's size for renamed files.
+        _fs.rename = function(oldPath, newPath, callback){
+            _innerRename.call(_fs, oldPath, newPath, function(err){
+                // only update cache once original was successfull
+                if (!err){
+                    _cache[newPath] = _cache[oldPath];
+                    delete _cache[oldPath];
+                }
+                callback(err);
+            });
+        };
+
+        // returns true/false depending on whether root/index.html exists
+        ProjectStats.hasIndexFile = function(){
+            if (!_root){
+                return false;
+            }
+
+            var index = _Path.join(_root, "index.html");
+            // Loop through all paths, return true if one of them is root/index.html
+            return !!(Object.keys(_cache)).find(function(path){
+                return path === index;
+            });
+        };
+
+        // returns bytes of files in the root
+        ProjectStats.getTotalProjectSize = function(){
+            // Sum up all sizes in the cache and return
+            if (!_root){
+                return 0;
+            }
+
+            return (Object.values(_cache)).reduce(function(acc, val){
+                return acc + (val ? val : 0);
+            });
+        };
+
+        // returns project file count
+        ProjectStats.getFileCount = function(){
+            if (!_root){
+                return 0;
+            }
+            return Object.keys(_cache).length;
+        };
+
+        // reinitialize project stats
+        _root = root;
+        _cache = {};
+
+        // walk the root
+        _shell.find(_root, { exec:addSize }, callback);
+    };
+
+    // accepts an array of mimes and/or file extensions
```

Comment:
I don't want to hold this PR up any longer, and adding new things to it is a bad idea (we've already taken way too long).  It's better to do small, incremental changes to a piece of code vs. trying to ram through a lot of things in one go.  To you, this is all related work, and should probably happen at the same time.  However, it's more efficient if we break the work up into separate PRs that build on each other.

Please remove this and start a new PR for this extra work.